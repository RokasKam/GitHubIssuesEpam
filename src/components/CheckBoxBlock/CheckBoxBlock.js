import React from 'react'
import { View, Text } from 'react-native';
import CheckBox from '@react-native-community/checkbox';
import { styles } from './CheckBoxBlock.style';

export const CheckBoxBlock = ({ text, value, setValue }) => {

    return (

        <View style={styles.checkBoxBlock}>
            <CheckBox
                value={value}
                onValueChange={() => {
                    setValue(!value)
                }}
                tintColors={{ true: "#00bf33", false: 'gray' }}
                onTintColor="#00bf33"
                onCheckColor="#00bf33"
            />
            <Text style={styles.text}>{text}</Text>
        </View>
    )
};

