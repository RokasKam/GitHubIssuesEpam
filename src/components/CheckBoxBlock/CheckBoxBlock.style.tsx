import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    checkBoxBlock: {
        flexDirection: "row",
        alignItems: "center",
        marginBottom: 20,

    },
    text: {
        marginLeft: 10,
        color: "#323C4C",
    }
});