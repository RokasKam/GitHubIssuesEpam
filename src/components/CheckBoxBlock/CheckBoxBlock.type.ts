export type Props = {
    text: string;
    value: boolean;
    setValue?: (value: boolean) => void;
};