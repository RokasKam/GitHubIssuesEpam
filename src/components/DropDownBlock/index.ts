import { DropDownBlock } from "./DropDownBlock";
export { DropDownBlock };