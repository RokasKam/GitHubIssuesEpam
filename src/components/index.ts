export { CheckBoxBlock } from "./CheckBoxBlock";
export { DropDownBlock } from "./DropDownBlock";
export { IssuesList } from "./IssuesList";
export { ListCard } from "./ListCard";
export { SearchButton } from "./SearchButton";
export { SearchInput } from "./SearchInput";

