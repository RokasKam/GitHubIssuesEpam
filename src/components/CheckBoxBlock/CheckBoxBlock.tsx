import React from 'react'
import { View, Text } from 'react-native';
import CheckBox from '@react-native-community/checkbox';
import { styles } from './CheckBoxBlock.style';
import { Props } from './CheckBoxBlock.type';

export const CheckBoxBlock: React.FC<Props> = ({ text, value, setValue }) => {

    return (

        <View style={styles.checkBoxBlock}>
            <CheckBox
                value={value}
                onValueChange={() => setValue?.(!value)}
                tintColors={{ true: "#00bf33", false: 'gray' }}
                onTintColor="#00bf33"
                onCheckColor="#00bf33"
            />
            <Text style={styles.text}>{text}</Text>
        </View>
    )
};

