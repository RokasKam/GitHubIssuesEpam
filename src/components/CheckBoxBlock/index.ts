import { CheckBoxBlock } from "./CheckBoxBlock";
export { CheckBoxBlock };