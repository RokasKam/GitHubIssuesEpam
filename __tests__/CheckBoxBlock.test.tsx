import React from 'react';
import renderer from 'react-test-renderer';
import { CheckBoxBlock } from '../src/components/CheckBoxBlock'

test('renders correctly', () => {
  const tree = renderer.create(<CheckBoxBlock text={''} value={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});