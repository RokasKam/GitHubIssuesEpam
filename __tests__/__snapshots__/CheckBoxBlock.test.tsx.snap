// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 20,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
  >
    <RNCCheckbox
      forwardedRef={null}
      onCheckColor="#00bf33"
      onTintColor="#00bf33"
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      tintColors={
        Object {
          "false": "gray",
          "true": "#00bf33",
        }
      }
      value={false}
    />
  </View>
  <Text
    style={
      Object {
        "color": "#323C4C",
        "marginLeft": 10,
      }
    }
  />
</View>
`;
